hare/haretypes.go

```diff
@@ -30,73 +31,96 @@ func NewBytes32(buff []byte) Bytes32 {
 	return x
 }
 
-func (b32 *Bytes32) Id() uint32 {
+func (b32 Bytes32) Id() uint32 {
 	h := fnv.New32()
 	h.Write(b32[:])
 	return h.Sum32()
 }
 
-func (b32 *Bytes32) Bytes() []byte {
+func (b32 Bytes32) Bytes() []byte {
 	return b32[:]
 }
 
 type Set struct {
-	blocks []BlockId
+	blocks map[uint32]BlockId
 }
 
-func NewSet(data [][]byte) Set {
-	s := Set{}
+func NewEmptySet() *Set {
+	s := &Set{}
+	s.blocks = make(map[uint32]BlockId, layerSize)
```

Comment:
You shouldn't use `layerSize` for several reasons - 
1. The Layer size isn't fixed, it is fixed only on avarage
2. You shouldn't mix domain terms with the internal hare protocol (that can be used outside of the scope of Spacmesh)
3. I very much agree that when allocating a slice you should do best effort to set the capacity, but in this case - since it is used not often (only by the leader)

Reply:
Fixed.